ucsd-cse125-sp22/cse125-sp22-group1
# Audience Interaction

This adds the ability to send out question prompts to all connected web clients to vote on what happens.

### How to test
- do `yarn dev` in the web folder to get the web client running, then go to localhost:3000 to see the app
- do `cargo run` in server and client to get the server running
- join active game via web ui
- vote for something
- wait 30 seconds
- see if your vote won!

### Things to note
- i think I created alot of different structures for this, would appreciate help in terms of how i might simplify stuff
- how would the methods i created flow with the server code / make it easy to send requests
- how i handled serialization; i used serde_json to make communicating with the web easier, but I wasn't able to get the same `trait` style implementation working that @KNOXDEV got working really nicely (would appreciate feedback / code on how to do that if you think it's nicer)
- Netlify builds aren't working yet, but that's because I'm tweaking stuff (vercel is nicer, but vercel doesn't have free deployment of organizations so i'm using netlify)
